### ISSUE TITLE
BAP opam-repository opam-2.0 support

### ISSUE THREAD

**Issue by XVilka** [BinaryAnalysisPlatform/bap#862]
Since opam-repository doesn't have issues enabled, opening it here. Opam 2.0 will be switched to the default format in a couple weeks: http://opam.ocaml.org/blog/opam-2-0-0-repo-upgrade-roadmap/
I use it with latest opam-2.0 available and have this warning.
```
[WARNING] The repository 'bap' at git+https://github.com/BinaryAnalysisPlatform/opam-repository.git#testing doesn't have a 'repo' file, and might not be compatible with this version of opam.
```

**Comment by gitoleg** [comment]
Recently, we decided to clear our `testing` branch from non-bap packages. 
The reason is that we don't want to update our repository any time the main one updated. And from a user perspective, adding a new repository with `bap` packages just extends the main repository, but not replaces anything. 
That's why we have less files in there, and that's why `repo` file is
missed. 
The only new behaviour here that `opam 1.2` doesn't complain about it, while
`opam 2.0` does. So indeed, it could be better to restore `repo` (or whatever else) to avoid such warnings.

**Comment by ivg** [comment]
we have released to opam (https://github.com/ocaml/opam-repository/pull/12874) so everything should work fine now. 

@gitoleg, make sure that you have updated the testing branch, as well as that we have the corresponding release branch in our repo.
